pkg/component/observability/monitoring/prometheus/cache/assets/scrapeconfigs/cadvisor.yaml

```diff
@@ -6,6 +6,7 @@ metrics_path: /metrics/cadvisor
 
 tls_config:
   ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+  insecure_skip_verify: true
```

Comment:
Is it possible to only set this if really needed, i.e. in the managed seed scenarios described in the commit message, or would that additional effort/complexity not be worth the effort? If the KEP you mentioned is already on the horizon and we will have a proper solution mid-term anyway I am also fine with this as the potential impact is quite limited.
WDYT?

Reply:
> Is it possible to only set this if really needed, i.e. in the managed seed scenarios described in the commit message

Unfortunately I'm not sure how to figure out if the seed is a managed seed.

The scrape configs are used (could be processed) here: https://github.com/gardener/gardener/blob/fa31ef3ee09b5d5fd15666617348c0ac3471f182/pkg/component/observability/monitoring/prometheus/cache/scrapeconfigs.go#L12-L15
and after tracing back the function call hierarchy, this is called by the seed reconciler here:
https://github.com/gardener/gardener/blob/fa31ef3ee09b5d5fd15666617348c0ac3471f182/pkg/gardenlet/controller/seed/seed/reconciler.go#L52
The `seed` object itself is available, but the information if it is a managed seed (i.e. the cluster is a shoot) or not (i.e. the cluster is a IaaS Kubernetes cluster) is not available in the seed.

Would it be idiomatic to use the garden client and try to look for a managed seed with the same name as the seed. If it exists, it is a managed seed, otherwise it isn't.
https://github.com/gardener/gardener/blob/fa31ef3ee09b5d5fd15666617348c0ac3471f182/pkg/gardenlet/controller/seed/seed/reconciler.go#L56-L62

Reply:
> use the garden client and try to look for a managed seed with the same name as the seed. If it exists, it is a managed seed, otherwise it isn't.

@ScheererJ I tried to implement the approach above and passed down the `isManagedSeed` flag to the `AdditionalScrapeConfigs` functions and used golang text templating to set the `insecure_skip_verify` flag.

I attempted to test this in the KinD based local setup. The first idea of faking the KinD cluster to be a managed seed by creating a ManagedSeed object for the KinD cluster failed (the various admission checks properly prevent that). Then, instead of using the default `local` project, I created the shoot in the `garden` project (namespace) and upgraded it to a managed seed with:

```yaml
apiVersion: seedmanagement.gardener.cloud/v1alpha1
kind: ManagedSeed
metadata:
  name: local
  namespace: garden
spec:
  shoot:
    name: local
  gardenlet:
    bootstrap: BootstrapToken 
    mergeWithParent: true
    config:
      seedConfig:
        spec:
          ingress: 
            controller:
              kind: nginx
```

This way I could confirm that

```
k get secrets prometheus-cache-additional-scrape-configs -o json | jq -r '
  .data."prometheus.yaml" | @base64d
' | grep skip
```

TLS verification is enabled in the KinD cluster but disabled in the managed seed.

It is nice that with the local setup we can upgrade a shoot to a seed:

```
k -n garden get seeds,shoots                                                                                                                                                                                                    remote-local-setup-0: Fri May 17 16:18:49 2024

NAME                                           STATUS   LAST OPERATION               PROVIDER   REGION   AGE   VERSION       K8S VERSION
seed.core.gardener.cloud/local                 Ready    Reconcile Succeeded (100%)   local      local    11m   v1.95.0-dev   v1.30.0
seed.core.gardener.cloud/local-ha-multi-zone   Ready    Reconcile Succeeded (100%)   local      local    76m   v1.95.0-dev   v1.29.2

NAME                              CLOUDPROFILE   PROVIDER   REGION   K8S VERSION   HIBERNATION   LAST OPERATION               STATUS    AGE
shoot.core.gardener.cloud/local   local          local      local    1.30.0        Awake         Reconcile Succeeded (100%)   healthy   71m
```

https://github.com/gardener/gardener/pull/9716/commits/05d926c55d4bfccab345ecb259bd8b30f879134e